### DISCUSSION TITLE
How to Access Asset Metadata Value Defined in @asset?

### DISCUSSION THREAD

**Discussion by dennisgera** [dagster-io/dagster#22568] [Q&A] [answered]
I need to access the metadata value that I defined using the metadata parameter in the @asset decorator. While I know how to access output metadata using the context.get_output_metadata() function, I haven’t been able to figure out a way to retrieve the metadata value that I specified in the @asset definition itself.

Here's an example of what I'm trying to do:

```python
@asset(
    metadata={
        "entity": "an_entity",
    }
)
def processed_data(context):
    entity = ... 
    return do_something_with_entity(entity)
```

In this example, I have defined an asset processed_data with a metadata value such as entity. I would like to access this metadata value within my asset.

Is there a specific function or method in Dagster that allows for retrieving these metadata values?

**Comment by garethbrickman** [comment, marked as answer]
You can access definition-level metadata from the `AssetExecutionContext`:

```python
from dagster import asset, AssetExecutionContext

@asset(
    metadata={
        "entity": "an_entity",
    }
)
def processed_data(context: AssetExecutionContext):
    # Access the metadata defined in the asset decorator
    asset_metadata = context.assets_def.metadata_by_key[context.asset_key]
    entity = asset_metadata["entity"]

    context.log.info(f"Entity: {entity}")

    return do_something_with_entity(entity)
```

**Reply by dennisgera** [reply]
Thanks!!
